docs/contributor/arch/021-refactor-performance-tests.md

```diff
@@ -0,0 +1,44 @@
+# 20. Refactor Performance Tests
+
+Date: 2025-07-03
+
+## Context
+
+Currently, [performance tests](../benchmarks/README.md) are written in bash which makes them hard to maintain. Our goal is to rewrite them in Golang so that they become easier to read, update and debug.
+
+### POC for using components from Opentelemtry Collector Testbed
+
+Since we are going to rewrite the performance tests in Golang, this is a good point of time to check if it is usefule to use components from the [Opentelemetry Collector Testbed](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/testbed).
+
+A [POC](../pocs/opentelemetry-testbed/) has been implemented for using the `LoadGenerator` and `MockBackend` from the Opentelemetry Collector Testbed. The performance tests in opentelemetry-collector-contrib repo are executed locally on a machine (all components run as processes and not containers). In our use case, we want test the images which will be running at the end as containers in production kubernetes clusters. So, in order to test our Telemetry module with the `LoadGenerator` and `MockBackend`, these 2 components are dockerized and running as pods in the POC . The setups in the POC for testing the Log Gateway and Log Agent are shown respectively in the diagrams below:
+
+![arch](./../assets/opentelemetry-testbed-log-gateway-setup.svg)
+
+![arch](./../assets/opentelemetry-testbed-log-agent-setup.svg)
+
+#### Advantages:
+- It is possible to determine the exact number of data items (trace spans, metric data points, or log records) sent by the load generator and the exact number of data items received by the mock backend. In the POC, these numbers are exposed as a prometheus metric at the `/metrics` endpoint on port `2112`.
+
+#### Disadvantages:
+- There is no [DataSender](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/testbed/datasenders) available that writes logs to stdout.
+    - The [FileLogWriter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/testbed/datasenders/stanza.go) converts OTLP logs to text lines and writes them to a temporary log file. This is not suitable for our Log Agent, since it tails logs which are only written to the stdout.
+    - The workaround in the POC is to write a custom `stdoutLogGenerator` which implements the `DataSender` interface. The `stdoutLogGenerator` is very similar to the `FileLogWriter`, but it writes logs to stdout instead of a temporary log file.
```

Comment:
```suggestion
    - The workaround in the POC is to write a custom `stdoutLogGenerator`, which implements the `DataSender` interface. The `stdoutLogGenerator` is very similar to the `FileLogWriter`, but it writes logs to stdout instead of a temporary log file.
```